src/QirTools/pyqir/pyqir/__init__.py

```diff
@@ -1,59 +1,214 @@
 
     def reset(self, target: str):
+        """
+        Applies Reset operation to the target qubit
+
+        :param target: name of the target qubit
+        :type target: str
+        """
         self.pyqir.reset(target)
 
     def rx(self, theta: float, qubit: str):
+        """
+        Applies Rx operation to the target qubit
+
+        :param theta: rotation value for target qubit
+        :type theta: float
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.rx(theta, qubit)
 
     def ry(self, theta: float, qubit: str):
+        """
+        Applies Ry operation to the target qubit
+
+        :param theta: rotation value for target qubit
+        :type theta: float
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.ry(theta, qubit)
 
     def rz(self, theta: float, qubit: str):
+        """
+        Applies Rz operation to the target qubit
+
+        :param theta: rotation value for target qubit
+        :type theta: float
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.rz(theta, qubit)
 
     def s(self, qubit: str):
+        """
+        Applies S operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.s(qubit)
 
     def s_adj(self, qubit: str):
+        """
+        Applies SAdj operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.s_adj(qubit)
 
     def t(self, qubit: str):
+        """
+        Applies T operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.t(qubit)
 
     def t_adj(self, qubit: str):
+        """
+        Applies TAdj operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.t_adj(qubit)
 
     def x(self, qubit: str):
+        """
+        Applies X operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.x(qubit)
 
     def y(self, qubit: str):
+        """
+        Applies Y operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.y(qubit)
 
     def z(self, qubit: str):
+        """
+        Applies Z operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.z(qubit)
 
-    def add_classical_register(self, name, size):
+    def add_classical_register(self, name: str, size: int):
+        """
+        Models a classical register of the given size. The individual values
+        are accessed by name "<name><index>" with 0 based indicies.
+        Example:
```

Comment:
Should this example be formatted as an RST code block?
```suggestion
        Example::

```